types/shallow-equals/index.d.ts

```diff
@@ -0,0 +1,14 @@
+// Type definitions for shallow-equals 1.0
+// Project: https://github.com/hughsk/shallow-equals
+// Definitions by: Ross Solomon <https://github.com/rsolomon>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = shallow_equals;
+
+declare function shallow_equals<A, B>(
+  a?: A,
+  b?: B,
+  compare?: (objA?: A, objB?: B) => boolean,
```

Comment:
As discussed [here](http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html) under "Optional Parameters in Callbacks", please don't make `objA` and `objB` optional.

Reply:
TIL